.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,54 @@
+name: Integration tests
+
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  get-head-commit-msg:
+    runs-on: ubuntu-latest
+    outputs:
+      head-commit-msg: ${{ steps.step-head-commit.outputs.headCommitMsg }}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Get Head Commit Message
+        id: step-head-commit
+        run: echo "headCommitMsg=$(git show -s --format=%s)" >> "$GITHUB_OUTPUT"
+
+  integration-tests:
+    runs-on: ubuntu-latest
+    needs: get-head-commit-msg
+    if: ${{ !contains(needs.get-head-commit-msg.outputs.head-commit-msg, 'ci-skip-integration-tests') }}
+    env:
```

Comment:
I think we can get rid of this env if we use it only to populate the defaults one line below.

Reply:
We rather have a `CI: "true"` variable in the original Gitlab file, do we need to add it here as well?

Reply:
Added `CI: "true"` , I will keep `working_dir` since `actions/setup-node@v4` didn't support it fully yet